### PR TITLE
chore: add og image

### PR DESCRIPTION
**Pull request by CinematicCow**
fix for #258 

added a static open graph image for now.

### CODE REVIEW

**Comment by orhun** [comment]
Nice! Do you know a way to test this locally before deploying it?

**Comment by kdheepak** [comment]
I had to create a local tunnel first, and spun up a dev version from this PR. Then I used this website to test: https://www.opengraph.xyz

I wasn't able to get it to work properly if the image was part of the repo. I was able to get it to work properly when I linked to an image explicitly.

image

image

**Comment by kdheepak** [comment]
ratatui-og

File: astro.config.mjs [hunk 1]
```diff
@@ -40,6 +40,15 @@ export default defineConfig({
         dark: "./src/assets/logo-dark.png",
       },
       favicon: "./src/assets/logo.png",
+      head:[
+        {
+          tag:"meta",
+          attrs:{
+            property:"og:image",
+            content:"./src/assets/ratatui-og.png"
```
Comment (kdheepak) [inline astro.config.mjs]:
```suggestion
            content: "https://private-user-images.githubusercontent.com/1813121/290281077-352abbe5-0032-44ba-aae7-0e5b3d5dff4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0ODk3NTIsIm5iZiI6MTcwMjQ4OTQ1MiwicGF0aCI6Ii8xODEzMTIxLzI5MDI4MTA3Ny0zNTJhYmJlNS0wMDMyLTQ0YmEtYWFlNy0wZTViM2Q1ZGZmNGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTNUMTc0NDEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWQ5YjExYzQwNzJiY2ZiNjYxOWI5ZmE4ZThjZTc5NDQ3NWI4ZjY0YmU1YjExNzg1NTY0NGJlNGU5OWE5ODJlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dvS3bJtJP2cnCXERlkRj3oY9nvn9UC7Ae6FIwbcAMFk"
```

File: src/assets/ratatui-og.png
Comment (kdheepak) [inline src/assets/ratatui-og.png:1]: If you accept the above suggestion, you can remove this file.

**Comment by joshka** [comment]
Inspired by this I updated the main page to just use that image as the hero image on the left - I think it looks pretty nice.

https://github.com/ratatui-org/website/pull/279

File: src/assets/ratatui-og.png
Comment (joshka) [reply src/assets/ratatui-og.png:1]: I'd prefer to have something that's more easily editable than a random github image link like this.

File: astro.config.mjs (hunk 1 above)
Comment (joshka) [reply astro.config.mjs]: I'd prefer to have something that's more easily editable than a random github image link like this.

Comment (kdheepak) [reply astro.config.mjs]: The problem is without this change astro adds this as the link in the `meta` tags: 

`https://ratatui.rs/./src/assets/ratatui-og.png`

**Comment by CinematicCow** [comment]
Fixed the relative path for the image. Also checked with ngrok tunnel on https://www.opengraph.xyz, it's working now.

**Comment by joshka** [comment]
Some quick observations:
- The proportions of negative space seem (subjectively) mis-weighted (I could be wrong on this, so please tell me if I am).
  - Left / right might be the same number of pixels, but the line / no line makes if seem unequal. Perhaps move things inward a little?
  - top / bottom definitely are unequal - does this cause any problems to if it is vertically centered? Does it look better to you?
  - 
On the website, I just changed things up to just "Ratatui" instead of Welcome to Ratatui!" and dropped the final period off the tagline. Do you think this would look good on the og:image?

**Comment by CinematicCow** [comment]
I see it now, it is not vertically aligned indeed. I quickly put this image together on figma. I'll make the required changes. The recommended image size is `1200x630px` according to opengraph. Do you want the box border around the logo or should I remove it? 

If we remove the "Welcome to ratatui" and replace it with "Ratatui" wouldn't the og:image look a little empty, as the required image size is somewhat big.

**Comment by kdheepak** [comment]
> Fixed the relative path for the image. Also checked with ngrok tunnel on https://www.opengraph.xyz/, it's working now.

@CinematicCow thanks for checking! Did you use `npm run dev`? Because it'll probably work with that but won't when it is deployed. 

You might have to try `npm run build && npm run preview`. I think this will only work if you add the image to the `public` folder and link it as `/ratatui-og.png`.

**Comment by CinematicCow** [comment]
- Changed the `og:image` to fix the spacing issues and added socials so it doesn't look empty. 
- Fixed the path for the image (thanks @kdheepak ), so now it works with build preview.

Here's how it looks now (screenshots taken on build preview)
image
image

**Review by orhun — APPROVED** [review]
LGTM

**Review by kdheepak — APPROVED** [review]

**Pull request merged**